src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const modified = Object.assign({}, state);
+  const resultArr = [];
+
+  for (let i = 0; i < actions.length; i++) {
+    switch (actions[i].type) {
+      case 'addProperties':
+        Object.assign(modified, actions[i].extraData);
+        break;
+
+      case 'removeProperties':
+        for (const item of actions[i].keysToRemove) {
+          delete modified[item];
+        }
+        break;
+
+      case 'clear':
+        for (const key of Object.keys(modified)) {
+          delete modified[key];
+        }
```

Comment:
```suggestion
        modified = {}
```

Reply:
strange, but before this option did not erase all the keys. 👎 
Therefore, I decided to manually erase them each through a cycle :-/